src/wire/igmpv2.rs

```diff
@@ -0,0 +1,417 @@
+    pub fn emit<T>(&self, packet: &mut Packet<&mut T>, checksum_caps: &ChecksumCapabilities)
+        where T: AsRef<[u8]> + AsMut<[u8]> + ?Sized
+    {
+        match self {
+            &Repr::MembershipQuery {
+                max_resp_time,
+                group_addr,
+                version
+            } => {
+                packet.set_msg_type(Message::MembershipQuery);
+                match version {
+                    IgmpVersion::Version1 =>
+                        packet.set_max_resp_time(0),
+                    IgmpVersion::Version2 =>
+                        packet.set_max_resp_time(max_resp_time.min(127) as u8),
+                }
+                packet.set_group_address(group_addr);
+            }
+            &Repr::MembershipReport {
+                group_addr,
+                version,
+            } => {
+                match version {
+                    IgmpVersion::Version1 => packet.set_msg_type(Message::MembershipReportV1),
+                    IgmpVersion::Version2 => packet.set_msg_type(Message::MembershipReportV2),
+                };
+                packet.set_max_resp_time(0);
+                packet.set_group_address(group_addr);
+            }
+            &Repr::LeaveGroup { group_addr } => {
+                packet.set_msg_type(Message::LeaveGroup);
+                packet.set_group_address(group_addr);
+            }
+        }
+        if checksum_caps.igmp.tx() {
+            packet.fill_checksum()
+        } else {
+            // make sure we get a consistently zeroed checksum,
+            // since implementations might rely on it
+            packet.set_checksum(0);
+        }
+    }
+}
+
+
+
+impl<'a, T: AsRef<[u8]> + ?Sized> fmt::Display for Packet<&'a T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match Repr::parse(self, &ChecksumCapabilities::default()) {
+            Ok(repr) => write!(f, "{}", repr),
+            Err(err) => write!(f, "IGMPv2 ({})", err),
+        }
+    }
+}
+
+impl<'a> fmt::Display for Repr {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            &Repr::MembershipQuery {
+                max_resp_time,
+                group_addr,
+                version,
+            } => {
+                write!(f,
+                       "IGMPv2 membership query max_resp_time={} group_addr={} version={:?}",
+                       max_resp_time,
+                       group_addr,
+                       version)
+            }
+            &Repr::MembershipReport {
+                group_addr,
+                version,
+            } => {
+                write!(f,
+                       "IGMPv2 Membership report group_addr={} version={:?}",
+                       group_addr,
+                       version)
+            }
+            &Repr::LeaveGroup { group_addr } => {
+                write!(f, "IGMPv2 Leave Group group_addr={})", group_addr)
```

Comment:
All three branches have inconsistent capitalization. I prefer `membership query`, `membership report` and `leave group`.